tests/integration/trace/test_zipkin_trace.py

```diff
@@ -176,3 +178,17 @@ def hello(request):
 
     yield f
     assert count[0] == 1
+
+
+@pytest.mark.gen_test
+def test_zipkin_trace_sampling(trace_server):
+    run_times = 100000
+    sample_rate = 1.0 - random.random()
+    hook = ZipkinTraceHook(sample_rate=sample_rate)
+    count = 0
+    for _ in range(run_times):
+        if hook._lucky(_uniq_id()):
+            count += 1
+
+    assert 0.9 * run_times * sample_rate <= count
```

Comment:
Yeah. I added the test for the first case.
For the second situation, the previous two tests that I set the sample_rate to 1, equals to make the lucky return True and verify the call been made.